lib/zero_silent_failures/manual_remediation/saved_claim.rb

```diff
@@ -0,0 +1,213 @@
+
+module ZeroSilentFailures
+  module ManualRemediation
+    # super class for manual remediation of a SavedClaim type
+    class SavedClaim
+      attr_reader :claim, :zipfile, :aws_download_zip_link
+
+      # constructor
+      #
+      # @param saved_claim_id [Integer] the id of the claim needing manual remediation
+      def initialize(saved_claim_id)
+        @claim = claim_class.find(saved_claim_id)
+        @zipfile = "#{Common::FileHelpers.random_file_path}.zip"
+      end
+
+      # process the _claim_ for remediation:
+      # - generate and stamp claim and attachment pdf
+      # - create a zipfile of generated pdf and metadata.json
+      # - upload zipfile to AWS S3 bucket (only if running on prod)
+      # - update database records for claim FormSubmissionAttempts
+      # - all generated files are deleted
+      def run
+        Rails.logger.info "Manual Remediation for #{claim.form_id} #{claim.id} started"
+
+        # reset in case there are multiple runs with the same instance
+        @files = []
+
+        package_claim
+        upload_documents
+        update_database
+
+        Rails.logger.info "Manual Remediation for #{claim.form_id} #{claim.id} Complete!"
+      ensure
+        Common::FileHelpers.delete_file_if_exists(zipfile) if on_prod?
+        files.each { |file| Common::FileHelpers.delete_file_if_exists(file[:path]) }
+      end
+
+      # generate metadata, claim.pdf, attachment.pdf, and zip
+      def package_claim
+        files << generate_metadata_json
+        files << generate_form_pdf
+
+        claim.persistent_attachments.each do |pa|
+          files << generate_attachment_pdf(pa)
+        end
+
+        zip_files
+      end
+
+      # upload documents to s3 bucket (only if running on prod)
+      def upload_documents
+        Rails.logger.info "Uploading documents - #{claim.form_id} #{claim.id}"
+
+        aws_upload_zipfile if on_prod?
+        # @todo ? upload to sharepoint directly ?
+
+        aws_download_zip_link || zipfile
+      end
+
+      # update FormSubmissionAttempt records for _claim_
+      def update_database
+        Rails.logger.info "Updating database - #{claim.form_id} #{claim.id}"
+
+        fs_ids = claim.form_submissions.map(&:id)
+        FormSubmissionAttempt.where(form_submission_id: fs_ids, aasm_state: 'failure')&.map(&:manual!)
+      end
+
+      private
+
+      # is the current environment production
+      def on_prod?
+        Settings.vsp_environment == 'production'
+      end
+
+      # claim class to be used
+      # - inheritor should override to specific type
+      def claim_class
+        ::SavedClaim
+      end
+
+      # array of files to be packaged
+      def files
+        @files ||= []
+      end
+
+      # assemble metadata for _claim_
+      # - inheritor should append to _super_ if needed
+      def generate_metadata
+        form = claim.parsed_form
+        address = form['claimantAddress'] || form['veteranAddress']
```

Comment:
Are `claimantAddress` and `veteranAddress` guaranteed to exist?

Reply:
as far as i know for the majority of forms yes, but that is also a case where a form would want/need to inherit and override this function